Implement Discord->Minecraft Chat Formatting.

TheCurle opened the pull request:
It's pretty fun to have your messages formatted in-game.

This allows people to send messages in discord with Legacy style chat formatting sequences, and it will show in-game appropriately.
A full list of sequences, for future reference:

   &0: color black
   &1: color dark_blue
   &2: color dark_green
   &3: color dark_aqua
   &4: color dark_red
   &5: color dark_purple
   &6: color gold
   &7: color gray
   &8: color dark_gray
   &9: color blue
   &a: color green
   &b: color aqua
   &c: color red
   &d: color light_purple
   &e: color yellow
   &f: color white
  
   &k: effect obfuscated (makes the letters scrambled and unreadable)
   &l: effect bold
   &m: effect strikethrough (a single thin line through the whole text)
   &n: effect underline
   &o: effect italic
   &r: effect reset (undoes all of the effect formatting)
   
There are two known "issues" with this implementation:
- Formatting cannot be chained; ie, you can't do &l&5 to get bold purple text.
- It looks like ass

They can both be fixed with time, but i wanted to get this code on screen for you to look at.

TheCurle commented:
Updated for RGB and English language colors.

zml2008 commented:
Can I encourage you to align your formatting system with other rich text-focused projects? (entirely unbiased, I know) I have [MiniMessage](https://docs.adventure.kyori.net/minimessage/format.html) which is used in a fair few places across the community, and while I don't expect you to directly depend on the library or anything, but I do think it would be neat if this format was somewhat aligned with what we do.

TheCurle commented:
I'm all for standardising formatting!

I just implemented this because it was the path of least resistance (and I've implemented something incredibly similar for a personal project).

Marked this PR as draft until I align it closer to MiniMessage.

sciwhiz12 commented:
As a side-note because MiniMessage was mentioned here, I do have some future plans to use [KyoriPowered's adventure library](https://github.com/KyoriPowered/adventure) alongside the [MCDiscordReserializer library by Vankka](https://github.com/Vankka/MCDiscordReserializer) for proper Minecraft-to-Discord rendering of formatting.
